docs/features/software-catalog/configuration.md

```diff
@@ -0,0 +1,60 @@
+---
+id: software-catalog-configuration
+title: Catalog Configuration
+---
+
+## Static Location Configuration
+
+To enable declarative catalog setups, it is possible to add locations to the
+catalog via [static configuration](../../conf/index.md). Locations are added to
+the catalog under the `catalog.locations` key, for example:
+
+```yaml
+catalog:
+  locations:
+    - type: github
+      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/artist-lookup-component.yaml
+```
+
+The locations added through static configuration can not be removed through the
+catalog locations API. To remove the locations, you have to remove them from the
+configuration.
+
+## Catalog Rules
+
+By default the catalog will only allow ingestion of entities with the kind
+`Component` and `API`. In order to allow entities of other kinds to be added,
+you need to add rules to the catalog. Rules are added either in a separate
+`catalog.rules` key, or added to statically configured locations.
+
+For example, given the following configuration:
+
+```yaml
+catalog:
+  rules:
+    - allow: [Component, API]
+    - allow: [System]
+      locations:
+        type: github
```

Comment:
this is perhaps slightly confusing, looking at rules.locations vs location with implicit rule :) am tempted to add a `target` here

Reply:
still humbly wondering if this shouldn't have been

```
rules:
  - kind: [Component, API]
    action: allow
  - kind: [System] 
    locationType: github
    action: allow
```

or turned a bit inside out:

```
rules:
  - allow:
      kind: [Component, API]
  - allow:
      kind: [System] 
      locationType: github
```

both for clarity and extensibility, but not gonna be insisting on it

Reply:
or to be able to reuse the same structure in two places:

```yaml
rules:
  - action: allow
    match:
      kind: ...
locations:
  - type: ...
    target: ...
    match:
     kind: ...
```

Reply:
Ah this totally shouldn't have been here at all, it's not implemented :p

I was thinking we might do something like 

```yaml
rules:
  allow:
  - entities:
    - kind: Component
    locations:
    - type: github
```

Basically the value of `allow` would be some kind of list of entity + location matchers. Tbh I'm not sure that's ever gonna be needed though, and hopefully we have some more use-cases to base that off of.

Reply:
Realized I didn't address your second comment :grin:

I tried that out, but was running into trouble with the matcher being specific to the action type, and that it also makes the assumption that we'd be matching on things at all.

The second example kinda solves the above problem, and is why I basically when with that. The current format is just a shorthand, and it should be simple enough to expand the configuration if needed. As mentioned in my previous comment I think we should wait with that though.